File: submissions/0458-poor-pigs/solution.java
public class Solution {
    public int poorPigs(int buckets, int minutesToDie, int minutesToTest) {
        if(buckets <= 1) {
            return 0;
        }
        int round = minutesToTest / minutesToDie;
        int maxPerRound = round+1;
        int current = 1;
        int n = 1;
        while (true) {
            current *= maxPerRound;
            if (current >= buckets) {
                return n;
            }
            n++;
        }
    }
}

File: submissions/0458-poor-pigs/solution.py
from math import log
class Solution:
    def poorPigs(self, buckets: int, minutesToDie: int, minutesToTest: int) -> int:
        if buckets == 1:
            return 0

        rounds = minutesToTest // minutesToDie

        pigs = 1
        while (rounds+1)**pigs < buckets:
            pigs += 1

        return pigs
